platform/darwin/src/MGLFeature.mm

```diff
@@ -177,6 +177,18 @@ - (NSDictionary *)geoJSONDictionary {
     return mbgl::Feature{geometry};
 }
 
+- (mbgl::FeatureCollection)mbglFeatureCollection {
+    mbgl::FeatureCollection featureCollection;
+    featureCollection.reserve(self.shapes.count);
+    for (id <MGLFeaturePrivate> feature in self.shapes) {
+        if ([feature isMemberOfClass:[MGLShapeCollectionFeature class]]) {
```

Comment:
Good point. We won't ever get here if the super class, `MGLShapeCollection`, is used since that is a geometry that'll be handled on a different track than what we are doing here for features.

I don't think we actually need to support nested feature collections since it [does not look like they are supported in geometry.hpp](https://github.com/mapbox/geometry.hpp/blob/master/include/mapbox/geometry/feature.hpp#L72)